src/server/api/routers/question.ts

```diff
@@ -41,50 +41,63 @@ export const questionRouter = createTRPCRouter({
     });
   }),
   getAllReducedInfinite: publicProcedure
-    .input(z.object({ cursor: z.string().optional(), limit: z.number().min(1).max(100).nullable(), titleFilter: z.string().optional() }))
-    .query(
-      async ({input, ctx}) => {
-        const limit = input.limit ?? 50;
-        const totalCount = await ctx.prismaMongo.question.count({
-          where: input.titleFilter ? {
-            OR: [
-              {
-                title: {
-                  contains: input.titleFilter
-                },
-              },
-              {
-                body: {
-                  contains: input.titleFilter
-                }
-              }
-            ]
-          } : undefined,
-        });
-        const items = await ctx.prismaMongo.question.findMany({
-          take: limit + 1,
-          where: input.titleFilter ? {
-            title: {
-              contains: input.titleFilter
+    .input(
+      z.object({
+        cursor: z.string().optional(),
+        limit: z.number().min(1).max(100).optional(),
+        titleFilter: z.string().optional(),
+        backwards: z.boolean().optional(),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      const { limit = 50, titleFilter, backwards, cursor } = input;
+      const totalCount = await ctx.prismaMongo.question.count({
+        where: titleFilter
+          ? {
+              OR: [
+                { title: { contains: titleFilter } },
+                { body: { contains: titleFilter } },
+              ],
             }
-          } : undefined,
-          cursor: input.cursor ? {
-            title: input.cursor
-          } : undefined,
-          orderBy: {
-            title: 'asc'
-          }
-        });
-        let nextCursor: string | undefined = undefined;
-        if (items.length === limit + 1) {
-          const top = items.pop();
-          nextCursor = top!.title;
-        }
-        return {
-          items, nextCursor, totalCount
-        };
+          : undefined,
+      });
+      const items = await ctx.prismaMongo.question.findMany({
+        take: backwards ? -limit : limit,
```

Comment:
Next cursor requires us to use limit + 1 tho. Not sure why we allowing backwards? Coz it can complicate nextcursor

Reply:
@Gabau We need it to allow the user to navigate backwards on the questions page.

Reply:
it's not really needed if we store the previous cursors though, which is what https://trpc.io/docs/client/react/useInfiniteQuery does. I think then the simplest fix would be to separate the query into one for the questions route (which does not use useInfiniiteQuery) and one for the collab route (which will be the same as before  this pr)